src/System.Buffers.Experimental/System/Buffers/BufferReader_search.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Sequences;
+
+namespace System.Buffers
+{
+    public static partial class BufferReaderExtensions
+    {
+        public static bool TryReadUntill<TSequence>(ref this BufferReader<TSequence> reader, out ReadOnlyBuffer bytes, byte delimiter)
+            where TSequence : ISequence<ReadOnlyMemory<byte>>
+        {
+            var position = reader.PositionOf(delimiter);
+            if (position == null)
+            {
+                bytes = default;
+                return false;
+            }
+            bytes = new ReadOnlyBuffer(reader.Position, position.Value);
```

Comment:
Still not correct.

Reply:
as we discussed, we need to make it correct, or remove the ctor.